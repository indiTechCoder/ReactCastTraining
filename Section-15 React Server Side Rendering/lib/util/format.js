export default class {
  static date(dateString) {
    return new Date(dateString).toDateString();
  }

  static dateTime(dateTimeString) {
    const dt = new Date(dateTimeString);
    return `${dt.toDateString()} ${dt.toLocaleTimeString()}`;
  }
}
