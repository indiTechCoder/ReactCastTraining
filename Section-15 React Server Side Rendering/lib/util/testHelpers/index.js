export { default as renderer } from 'react-test-renderer';
export { shallow, mount } from 'enzyme';
export { default as enzymeHelper } from './enzymeHelper';
export { default as snapshotHelper } from './snapshotHelper';
