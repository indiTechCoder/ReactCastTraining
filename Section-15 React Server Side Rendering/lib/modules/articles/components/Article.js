import React from 'react';
import PropTypes from 'prop-types';
import glamorous from 'glamorous';
import storeProvider from '../../../app/storeProvider';
import format from '../../../util/format';

const StyledArticle = glamorous.div({
  paddingBottom: 10,
  borderBottomStyle: 'solid',
  borderBottomColor: '#aaa',
  borderBottomWidth: 1,
  marginBottom: 10
});

const ArticleTitle = glamorous.div({
  fontWeight: 'bold'
});

const ArticleDate = glamorous.div({
  fontSize: '0.85em',
  color: '#888'
});

const Author = glamorous.div({
  paddingTop: 10,
  paddingBottom: 10
});

const ArticleBody = glamorous.div({
  paddingLeft: 20
});

export const Article = ({ article, author }) => {
  return (
    <StyledArticle>
      <ArticleTitle>{article.title}</ArticleTitle>
      <ArticleDate>{format.date(article.date)}</ArticleDate>
      <Author>
        <a href={author.website} target="_blank">
          {author.firstName} {author.lastName}
        </a>
      </Author>
      <ArticleBody>{article.body}</ArticleBody>
    </StyledArticle>
  );
};

Article.propTypes = {
  article: PropTypes.shape({
    date: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    body: PropTypes.string.isRequired
  }),
  author: PropTypes.shape({
    firstName: PropTypes.string,
    lastName: PropTypes.string,
    website: PropTypes.string
  })
};

function extraProps(store, originalProps) {
  return {
    author: store.lookUpAuthor(originalProps.article.authorId)
  };
}

export default storeProvider(extraProps)(Article);
