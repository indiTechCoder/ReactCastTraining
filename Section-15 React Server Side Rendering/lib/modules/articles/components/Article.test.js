import React from 'react';
import { snapshotHelper } from '../../../util/testHelpers';
import { Article } from './Article';

const testAuthor = {
  website: 'testWebsite',
  firstName: 'testFirstName',
  lastName: 'testLastName'
};

const testProps = {
  article: {
    title: 'testTitle',
    date: 'Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)',
    body: 'testBody'
  },
  author: testAuthor
};

describe('Article', () => {
  it('renders correctly', () => {
    snapshotHelper.assertMatch(<Article {...testProps} />);
  });
});
