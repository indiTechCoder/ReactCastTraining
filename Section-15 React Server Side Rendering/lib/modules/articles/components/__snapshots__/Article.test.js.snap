// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article renders correctly 1`] = `
<div
  className="css-2hwute"
>
  <div
    className="css-l6c9n4"
  >
    testTitle
  </div>
  <div
    className="css-ilk9p0"
  >
    Sun Jul 10 2016
  </div>
  <div
    className="css-1pj8n87"
  >
    <a
      href="testWebsite"
      target="_blank"
    >
      testFirstName
       
      testLastName
    </a>
  </div>
  <div
    className="css-wwoukl"
  >
    testBody
  </div>
</div>
`;
